source/hpack/hpack.d

```diff
@@ -1,3 +1,115 @@
 //module vibe.http.internal.hpack.hpack;
 module hpack.hpack;
 
+import hpack.encoder;
+import hpack.decoder;
+import hpack.tables;
+
+import std.range;
+import std.typecons;
+import std.array; // appender
+import std.algorithm.iteration;
+
+void encodeHPACK(I,R)(I src, ref R dst, ref IndexingTable table, bool huffman = true)
+@safe
+{
+	static if(is(I == HTTP2HeaderTableField)) {
+		src.encode(dst, table, huffman);
+	} else if(is(ElementType!I : HTTP2HeaderTableField)){
+		src.each!(h => h.encode(dst, table, huffman));
+	}
```

Comment:
Should have an `else static assert (false, "...");` here to avoid silent misuse, or this could just be split up into two overloads, one having the `is(ElementType!I : HTTP2HeaderTableField)` as the template constraint.

Reply:
I added a template constraint, I'm merging this right now and I'll propose another PR with the final structure of the encoder. Thanks!